pkg/charts/linkerd2/addons.go

```diff
@@ -35,5 +27,21 @@ func ParseAddOnValues(values *Values) ([]AddOn, error) {
 		}
 	}
 
+	if values.Prometheus != nil {
+		if enabled, ok := values.Prometheus["enabled"].(bool); !ok {
+			return nil, fmt.Errorf("invalid value for 'prometheus.enabled' (should be boolean): %s", values.Prometheus["enabled"])
```

Comment:
If your type is an interface here, and it fails to cast to bool, wouldn't  you want to format it with %v instead?